src/main/java/seedu/duke/data/notebook/Notebook.java

```diff
@@ -34,10 +40,140 @@ public Notebook(ArrayList<Note> notes) {
+
+        if (notebookToggle == null) {
+            // Takes the notes in the notebook and sorts them according to title, alphabetically (a-z)
+            sortedNotes = (ArrayList<Note>) notes.stream()
+                    .filter(Objects::nonNull)
+                    .sorted(Comparator.comparing(a -> a.getTitle().toLowerCase()))
+                    .collect(Collectors.toList());
+        } else if (!notebookToggle) {
+            sortedNotes = (ArrayList<Note>) notes.stream()
+                    .filter(Objects::nonNull)
+                    .filter((s) -> !s.getPinned())
+                    .sorted(Comparator.comparing(a -> a.getTitle().toLowerCase()))
+                    .collect(Collectors.toList());
+        } else if (notebookToggle) {
+            sortedNotes = (ArrayList<Note>) notes.stream()
+                    .filter(Objects::nonNull)
+                    .filter((s) -> s.getPinned())
+                    .sorted(Comparator.comparing(a -> a.getTitle().toLowerCase()))
+                    .collect(Collectors.toList());
+        }
+
+        if (!isAscendingOrder) {
+            Collections.reverse(sortedNotes);
+        }
+
+        return sortedNotes;
+    }
+
+    /**
+     * Method compiles the ArrayList items and appends the items to a String.
+     * The ArrayList has already been sorted
+     * Method returns either top to bottom or bottom to top to account for ascending/descending sorting
+     *
+     * @return noteString String containing the notes sorted either ascending ot descending
+     */
+    public ArrayList<Note> getSortedList(Boolean isAscendingOrder, Boolean isPinned, ArrayList<Note> noteArrayList) {
+        ArrayList<Note> sortedNotes = new ArrayList<>();
+
+        if (isPinned) {
+            sortedNotes = (ArrayList<Note>) noteArrayList.stream()
+                    .filter(Objects::nonNull)
+                    .filter((s) -> s.getPinned())
+                    .sorted(Comparator.comparing(a -> a.getTitle().toLowerCase()))
+                    .collect(Collectors.toList());
+        } else if (!isPinned) {
+            sortedNotes = (ArrayList<Note>) noteArrayList.stream()
+                    .filter(Objects::nonNull)
+                    .filter((s) -> !s.getPinned())
+                    .sorted(Comparator.comparing(a -> a.getTitle().toLowerCase()))
+                    .collect(Collectors.toList());
+        } else if (isPinned == null) {
+            sortedNotes = (ArrayList<Note>) noteArrayList.stream()
+                    .filter(Objects::nonNull)
+                    .sorted(Comparator.comparing(a -> a.getTitle().toLowerCase()))
+                    .collect(Collectors.toList());
+        }
+
+        if (!isAscendingOrder) {
+            Collections.reverse(sortedNotes);
+        }
+
+        return sortedNotes;
+    }
+
+
+
     public Note getNote(int index) {
         return notes.get(index);
     }
 
+    public Note getNote(String noteTitle, ArrayList<Note> notebook) {
```

Comment:
As discussed, change the parameter ArrayList<Note> to take in a boolean fromOriginalNoteList or something similar and do a if else statement to determine which arrayList to get the note from. Also, consider the case that the note does not exist, what will the stream return?